src/porepy/numerics/ad/operators.py

```diff
@@ -1175,30 +1175,29 @@ def __init__(
 
         """
 
-        self._time_step_index: int = 0
+        self._time_step_index: int = -1
         """Time step index, starting with 0 (current time) and increasing for previous
         time steps."""
 
     @property
     def is_previous_time(self) -> bool:
         """True, if the operator represents a previous time-step."""
-        return True if self._time_step_index > 0 else False
+        return True if self._time_step_index >= 0 else False
 
     @property
-    def time_step_index(self) -> int:
+    def time_step_index(self) -> int | None:
         """Returns the time step index this instance represents.
 
-        - 0 indicates this is an operator at the current time step
-        - 1 represents the first previous time step
-        - 2 represents the next time step further back in time
+        - None indicates the current time (unknown value)
+        - 0 indicates this is an operator at the first previous time step
+        - 1 at the time step before
         - ...
 
-        Note:
-            Time-dependent operators with time step index 0 will always return the
-            value at the most recent iterate.
-
         """
-        return self._time_step_index
+        if self._time_step_index < 0:
+            return None
+        else:
+            return self._time_step_index
 
     def previous_timestep(
```

Comment:
What do you think about renaming these methods to the more precise `value_at_timestep/iteration(self, step)`?

Reply:
I believe you had some opinions on this, @keileg and @jwboth. Could you please comment?

Reply:
Works for me. What about the properties `is_previous_time` and `is_previous_iterate`? Do we get rid of "previous" there as well? Any suggestions for alternatives?

Reply:
> Works for me. What about the properties `is_previous_time` and `is_previous_iterate`? Do we get rid of "previous" there as well? Any suggestions for alternatives?

This is more difficult. I like the `is` prefix, indicating the boolean return type, but modifying the rest is difficult. One possibility is `is_lagged_in_time` (resp. `_iterate`), but that is not clearly better than using `previous` (so, switching to a synonym does not work). 

I think I'm blank, so unless other suggestions come up, I vote for keeping `is_previous_` (for now).

Reply:
I have not much too add. I appreciate getting rid of previous when fetching values.

Reply:
`.previous_timestep(steps: int = 1) -> .at_timestep(steps: int = 1)`

Using *previous* reflects that the implementation is going *backwards* in time, with corresponding number of steps.
The naming `at_timestep` indicates that the user could use a concrete index as an argument to "jump" between time steps.
That functionality would require some more safety checks.

Analogous for iterates.

Reply:
Regarding the safety checks about stepping:

As of now, the user could create "holes" in the sequence of prev time and iter.
I.e. Operators for index 0, 1, 3,4 for example, but not for index 2.
Is this critical? Or does it fall in the category "The user should know what he is doing"?

Reply:
I don't understand what you mean about the "holes". Please elaborate why this could be an issue, in person if you prefer.

Reply:
Discussed in person, issue deemed not critical.

Reply:
Jakub and I suggest `is_at_timestep` for the booleans.

Reply:
I am not sure how that is interpreted as a boolean property. Are you inquiring for a specific time step?